### PR TITLE
Support passing options to hackfmt

### PR DESCRIPTION
**Pull request by mwildehahn**
Initial pass at: https://github.com/hhvm/hack-codegen/issues/88

### CODE REVIEW

**Comment by mwildehahn** [comment]
@fredemmott updated based on your feedback.

File: src/BaseCodeBuilder.php [hunk 1]
```diff
@@ -111,8 +111,12 @@
     // if we're in a new line, insert indentation
     if ($this->isNewLine) {
       if ($this->indentationLevel !== 0) {
-        $n = $this->config->getSpacesPerIndentation() * $this->indentationLevel;
-        $this->code->append(\str_repeat(' ', $n));
+        if ($this->config->shouldUseTabs()){
+          $this->code->append(\str_repeat("\t", $this->indentationLevel));
+        } else {
+          $n = $this->config->getSpacesPerIndentation() * $this->indentationLevel;
+          $this->code->append(\str_repeat(' ', $n));
```
Comment (fredemmott) [inline src/BaseCodeBuilder.php]: `Str\repeat` from the HSL

File: test/CodegenFileTest.php [hunk 2]
```diff
@@ -384,13 +386,101 @@ public function testFormattingFullyGeneratedFile(): void {
...
+          ->addParameter(
+            'string $'.\str_repeat('b', 60),
+          )
+          ->setReturnType('(string, string)')
+          ->setBody(
+            $cgf->codegenHackBuilder()
+              ->addReturnf(
+                'tuple($%s, $%s)',
+                \str_repeat('a', 60),
+                \str_repeat('b', 60),
```
Comment (fredemmott) [inline test/CodegenFileTest.php]: same throughout

File: test/CodegenFileTest.php [hunk 3]
```diff
@@ -482,3 +574,54 @@ public function testConstants(): void {
...
+  }
+
+  public function shouldUseTabs(): bool {
+    return true;
+  }
+
+  public function getRootDir(): string {
+    return __DIR__;
+  }
+public function getFormatter(): ?ICodegenFormatter {
```
Comment (fredemmott) [inline test/CodegenFileTest.php]: run hackfmt on this file

**Pull request merged**